step-release-vis/src/app/components/environment/environment.ts

```diff
@@ -221,41 +221,54 @@ export class EnvironmentComponent implements OnInit, OnChanges {
     return 'middle';
   }
 
+  /* Returns the distance in pixels from the svg border. */
+  getPositionFromTimestamp(time: number): number {
+    return this.candidateService.scale(
+      time,
+      this.startTimestamp,
+      this.endTimestamp,
+      0,
+      this.svgWidth
+    );
+  }
+
   /*
-   * Computes the closes snapshot to the current position of the mouse.
-   *
+   * Computes the closest snapshot to the current position of the mouse.
```

Comment:
can we have a test with 20+ environments and the mouse being over the tooltip, but not over the data zone of the graph anymore? I really want to see that working from the user's perspective (see my comments on #240 wrt the same)

Reply:
@ancamaria99 is taking care of the tests for the tooltip when it is fixed by the mouse (did you mean 2+ environments?). After the tooltip has been fixed, the functions depending on the mouse are not used.

Reply:
ack for now, I will review things next week in the master where i see gaps ;)